lib/src/Pos/Communication/Limits.hs

```diff
@@ -310,6 +310,17 @@ instance (HasAdoptedBlockVersionData m, Applicative m) => MessageLimited MainBlo
     -- FIXME Integer -> Word32
     getMsgLenLimit _ = Limit . fromIntegral <$> maxBlockSize
 
+instance ( HasAdoptedBlockVersionData m
+         , Applicative m
+         )
+         => MessageLimited BlockHeader m where
+    getMsgLenLimit _ = f <$> getMsgLenLimit Proxy <*> getMsgLenLimit Proxy
+      where
+      maxLimit (Limit l1) (Limit l2) = Limit (max l1 l2)
+      f limA limB = 1 +
```

Comment:
Why `1`? The overhead is two bytes – one for the array, another for the tag. See [how `[1, ...]` is encoded](http://cbor.me/?diag=[1,%20%22Genesis%22]).

Reply:
I based this on the instance for `Either`:

```
instance ( MessageLimited a m
         , MessageLimited b m
         , Applicative m
         )
         => MessageLimited (Either a b) m where
    getMsgLenLimit _ = f <$> getMsgLenLimit Proxy <*> getMsgLenLimit Proxy
      where
      f limA limB = 1 + max (Left <$> limA) (Right <$> limB)
```

See `Pos.Communication.Limits.Types`. If this doesn't sound right to you, both instances should be fixed.

Reply:
cc @martoon what do you think?